frontend/src/lib/components/BranchButton.svelte

```diff
@@ -0,0 +1,384 @@
+					type: ToastType.Error,
+					dismissible: true
+				});
+			});
+			return; // Exit if response is not OK
+		}
+
+		// Extract and set the branches if the response is successful
+		response.json().then((data) => {
+			if (data.data && data.data.branches) {
+				// Map through branches to filter out protected ones and extract branch names
+				nonProtectedBranches = data.data.branches
+					.filter((branch: string) => !branch.includes('(protected)')) // Filter out protected branches
+					.map((branch: string) => branch.split(' (')[0]); // Extract just branch names
+				protectedBranches = data.data.branches
+					.filter((branch: string) => branch.includes('(protected)')) // Filter out protected branches
+					.map((branch: string) => branch.split(' (')[0]); // Extract just branch names
+			} else {
+				protectedBranches = []; // Reset if no branches found
+				nonProtectedBranches = [];
+			}
+		});
+	}
+
+	onMount(() => {
+		fetchExistingBranches();
+	});
+
+	async function setBranchName(input: string) {
+		// Define validation rules
+		const maxLength = 255; // Maximum length for branch name
+		const invalidCharacters = /[~^:?*<>|]/; // Invalid special characters
+		const startsWithLetterOrNumber = /^[a-zA-Z0-9]/; // Starts with letter or number
+		const containsSpaces = /\s/; // Contains spaces
+
+		// Define an array of rules
+		const rules = [
+			'Branch names must start with a letter or a number.',
+			'Branch names cannot contain spaces. Use dashes (-) or underscores (_) instead.',
+			'Branch names cannot contain special characters like ~, ^, :, ?, *, and others.',
+			`Branch names must be shorter than ${maxLength} characters.`
+		];
+
+		const isValidBranchName = (name: string) => {
+			return (
+				!containsSpaces.test(name) &&
+				startsWithLetterOrNumber.test(name) &&
+				!invalidCharacters.test(name) &&
+				name.length <= maxLength
+			);
+		};
+
+		if (!input) return;
+
+		// Validate branch name
+		if (!isValidBranchName(input)) {
+			addToast({
+				message: 'Please ensure your branch name follows these rules:\n' + rules.join('\n'),
+				type: ToastType.Warning,
+				dismissible: true
+			});
+			return;
+		}
+
+		if (protectedBranches.includes(input)) {
+			addToast({
+				message:
+					'Please select an existing branch name from the list of non-protected branches.\n You can also create your own',
+				type: ToastType.Warning,
+				dismissible: true,
+				timeout: 1500
+			});
+			return;
+		}
+
+		// Set branch name and reset state
+		branchName.set(input);
+		newBranchName = '';
+		showMenu = false;
+
+		if (!nonProtectedBranches.includes(input)) {
+			addToast({
+				message: `Changed branch name to "${input}".`,
+				type: ToastType.Success,
+				dismissible: true,
+				timeout: 1500
+			});
+			return;
+		}
+
+		// Call backend to update working directory by checking out the branch
+		const response = await fetch(`${apiAddress}/api/check-out/branches/${input}`, {
```

Comment:
```suggestion
		const response = await fetch(`${apiAddress}/api/checkout/branches/${input}`, {
```
Associated with the other comment about replacing `check-out` with `checkout`